docs/dev/optimizer.md

```diff
@@ -0,0 +1,55 @@
+# Optimizing Crypto Parameters
+
+`concrete-optimizer` is a tool that selects appropriate technical quantities for a given fully homomorphic encryption (FHE) calculation. These quantities have an impact on the security, correctness, and efficiency of the calculation.
+
+The security of the calculation is guaranteed to a required level, typically 128 bits. The correctness of the calculation is guaranteed up to a given error rate. The efficiency is optimized to reduce the time to execute the FHE calculus.
+
+These quantities, known as crypto-parameters, only exist due to the FHE nature of the calculus. The crypto-parameters are in a finite but huge space. The role of the optimizer is to quickly find the most efficient crypto-parameters possible while guaranteeing security and correctness.
+
+## Security, Correctness, and Efficiency
+
+### Security
+
+The security level is chosen by the user.
+
+We typically operate at a fixed security level, such as 128 bits, to ensure that there is never a trade-off between security and efficiency. This constraint imposes a minimum amount of noise in any calculation and in public material.
+
+An independent public research tool, the lattice estimator, estimates the security level.
+
+The lattice estimator is a joint work from FHE experts. It estimates the security level of the couple noise and the mask size of the ciphertext (i.e. one of the crypto-parameters).
+
+For each security level, a parameter curve of the appropriate minimal error level is pre-computed using the lattice estimator and used by the optimizer. Learn more about the parameter curve [here](https://www.notion.so/tools/parameter-curves/README.rst).
+
+### Correctness
+
+Correctness decreases as the level of noise increases.
+
+Noise accumulates during calculations until it is actively reduced. Too much noise can render the calculation's result inaccurate or, worse, completely incorrect.
+
+Before optimization, we compute a noise bound that guarantees a given error level if the noise always stays under that bound before we reduce it or before the result is decrypted. The noise growth depends on a critical quantity: the norm2 of any dot product (or equivalent) present in the calculus. This norm2 changes the scale of the noise, so we must reduce it sufficiently for the next dot product operation whenever we reduce the noise.
```

Comment:
> This norm2 changes the scale of the noise
During a dot product, the variance of the noise is multiplied by the square of the 2norm.